pkg/querier/stats_renderer.go

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package querier
+
+import (
+	"context"
+
+	promql_stats "github.com/prometheus/prometheus/util/stats"
+	prom_api "github.com/prometheus/prometheus/web/api/v1"
+
+	"github.com/grafana/mimir/pkg/querier/stats"
+)
+
+func StatsRenderer(ctx context.Context, s *promql_stats.Statistics, param string) promql_stats.QueryStats {
```

Comment:
This looks PromQL engine specific. Be aware we're migrating to Mimir Query Engine (MQE) and we need the tracking to work for MQE too. You can talk to @charleskorn about it.

Reply:
This is something that is not currently supported by MQE.

Is your goal to measure the total number of samples read from ingesters and store-gateways for the query?

Reply:
> Is the goal to measure the total number of samples read from ingesters and store-gateways for the query?

I would say this one.